src/GCHP_GridComp/GCHPctmEnv_GridComp/GCHPctmEnv_GridCompMod.F90

```diff
@@ -1,668 +1,460 @@
       type(ESMF_Clock),    intent(inout) :: CLOCK  ! The clock
-!
-! !OUTPUT PARAMETERS:
-      integer, optional,   intent(  out) :: RC     ! Error code
-!
-! !DESCRIPTION: 
-! The Run method of the derived variables CTM Gridded Component.
-!
-!EOP
-!-------------------------------------------------------------------------
-!BOC 
-!
-! !LOCAL VARIABLES:
-      character(len=ESMF_MAXSTR)      :: IAm = "Run"
-      integer                         :: STATUS
-      character(len=ESMF_MAXSTR)      :: COMP_NAME
-      type (MAPL_MetaComp), pointer   :: ggState
-      type (ESMF_Grid)                :: esmfGrid
-      type (ESMF_State)               :: INTERNAL
-      
-      ! Imports
-      real, pointer, dimension(:,:)   :: PS0 => null()
-      real, pointer, dimension(:,:)   :: PS1 => null()
-      real, pointer, dimension(:,:,:) :: SPHU0_IMPORT => null()
-      real, pointer, dimension(:,:,:) :: SPHU1_IMPORT => null()
-      real, pointer, dimension(:,:,:) :: UA  => null()
-      real, pointer, dimension(:,:,:) :: VA  => null()
-      real, pointer, dimension(:,:,:) :: temp3_r4  => null()
-      
-      ! Exports
-      real(r8), pointer, dimension(:,:,:) :: CX => null()
-      real(r8), pointer, dimension(:,:,:) :: CY => null()
-      real(r8), pointer, dimension(:,:,:) :: PLE1r8 => null()
-      real(r8), pointer, dimension(:,:,:) :: PLE0r8 => null()
-      real(r8), pointer, dimension(:,:,:) :: DryPLE1r8 => null()
-      real(r8), pointer, dimension(:,:,:) :: DryPLE0r8 => null()
-      real(r8), pointer, dimension(:,:,:) :: MFX => null()
-      real(r8), pointer, dimension(:,:,:) :: MFY => null() 
-      real(r8), pointer, dimension(:,:,:) :: SPHU0r8 => null()
-      real(r8), pointer, dimension(:,:,:) :: UpwardsMassFlux => null()
+      integer, optional,   intent(out) :: RC       ! Error code
 
       ! Locals
-      real, pointer, dimension(:,:,:) :: SPHU0 => null()
-      real, pointer, dimension(:,:,:) :: SPHU1 => null()
-      real, pointer, dimension(:,:,:) :: UC => null()
-      real, pointer, dimension(:,:,:) :: VC => null()
-      real(r8), pointer, dimension(:,:,:) :: PLEr8 => null()
-      real(r8), pointer, dimension(:,:,:) :: UCr8 => null()
-      real(r8), pointer, dimension(:,:,:) :: VCr8 => null()
-      
-      integer               :: km, k, is, ie, js, je, lm, l, ik
-      integer               :: ndt, isd, ied, jsd, jed
-      real(r8), allocatable :: AP(:), BP(:)
-      real(r8)              :: dt
-
-      ! Dry pressure calculations
-      integer               :: i, j
-      real(r8)              :: PSDry0, PSDry1, PEdge_Bot, PEdge_Top
-
-      logical, save         :: firstRun = .true.
+      character(len=ESMF_MAXSTR) :: IAm = "Run"
+      integer :: STATUS
+      character(len=ESMF_MAXSTR) :: COMP_NAME
+      type (MAPL_MetaComp), pointer :: ggState
+      type (ESMF_Grid) :: esmfGrid
 
+      ! Locals
+      real(r8), pointer  :: PLE(:,:,:) ! Edge pressures
```

Comment:
It takes some digging to figure out this is a pointer to PLE0 export. Maybe the name could be more clear on this?

Reply:
This is simply to pass PLE from `prepare_ple_exports` to `prepare_massflux_exports` for converting wind to mass flux. When the Env grid comp runs, PLE0 is the only PLE that's defined since Adv hasn't run.